main/static/js/map.js

```diff
@@ -0,0 +1,154 @@
+  function getPoints() {
+    return new Promise(function (resolve) {
+      resolve(map_data)
+    });
+  }
+
+  function createPopup(hospital) {
+    var mapPopup = document.getElementById("id_map_popup").content.cloneNode(true);
+
+    hosp = mapPopup.getElementById("id_hospital_name");
+    hosp.textContent = hospital.name;
+
+    link = mapPopup.getElementById("id_hospital_link");
+    link.href = hospital.link;
+
+    supply = mapPopup.getElementById("id_supply");
+    supply.textContent = hospital.supply_count;
+
+    repair = mapPopup.getElementById("id_repair");
+    repair.textContent = hospital.repair_count;
+
+    return mapPopup;
+  }
+
+  // initialize Leaflet
+  var map = L.map('id_map').fitWorld();
+
+  var myMarker;
+
+  var hospitalIcon = L.icon({
+      iconUrl: '/static/images/hospital-pin-black.svg',
+
+      iconSize:     [32, 39], // size of the icon
+      iconAnchor:   [16, 39], // point of the icon which will correspond to marker's location
+      popupAnchor:  [0, -42]  // point from which the popup should open relative to the iconAnchor
+  });
+
+  var hospitalBlueIcon = L.icon({
+      iconUrl: '/static/images/hospital-pin-blue.png',
+
+      iconSize:     [32, 39], // size of the icon
+      iconAnchor:   [16, 39], // point of the icon which will correspond to marker's location
+      popupAnchor:  [0, -42]  // point from which the popup should open relative to the iconAnchor
+  });
+
+  var hospitalGroup = L.featureGroup().addTo(map);
+  var hospitalArray = new Array();
+  var nearestGroup  = L.featureGroup().addTo(map);
+
+  map.locate({setView: true, maxZoom: 16});
+
+  function onLocationFound(e) {
+    if(myMarker == null) {
+      myMarker = L.marker(e.latlng).addTo(map);
+    }
+
+    findNearestMarker();
+  }
+
+  map.on('locationfound', onLocationFound);
+
+  function onLocationError(e) {
+    alert(e.message);
+    fitBounds();
+  }
+
+  map.on('locationerror', onLocationError);
+
+  // add the OpenStreetMap tiles
+  L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+    maxZoom: 19,
+    attribution: '&copy; <a href="https://openstreetmap.org/copyright">OpenStreetMap contributors</a>'
+  }).addTo(map);
+
+  // show the scale bar on the lower left corner
+  L.control.scale().addTo(map);
+
+  L.easyButton('<img class="h-5 inline" src="/static/images/hospital-pin-black.svg">', function(btn, map){
+    fitBounds();
+  }, 'Zoom all').addTo(map);
+
+  function locate() {
+    if (!navigator.geolocation){
+        alert("<p>Sorry, your browser does not support Geolocation</p>");
+        return;
+    }
+
+    map.locate({setView: true, maxZoom: 16});
+  };
+
+  L.easyButton('<span class="font-bold text-lg align-middle">&ofcir;</span>', function(btn, map){
+    locate();
+  }, 'Find your location').addTo(map);
+
+
+  function fitBounds () {
+    map.fitBounds(hospitalGroup.getBounds());
+  }
+
+  // fit hospitalGroup after 1 second.
+  setTimeout(function () {
+    //fitBounds();
+  }, 1);
```

Comment:
This can be removed

Reply:
This is a work in progress. Next release there should be other improvements/cleanup.